crates/roc_host/src/roc.rs

```diff
@@ -945,3 +886,89 @@ pub extern "C" fn roc_fx_tempDir() -> RocList<u8> {
 
     RocList::from(path_os_string_bytes.as_slice())
 }
+
+pub struct RocServer {
+    pub model: roc_std::SendSafeRocList<u8>,
+    pub captures: roc_std::SendSafeRocList<u8>,
+}
+
+unsafe impl Sync for RocServer {}
+
+pub fn call_roc_init() -> RocServer {
+    extern "C" {
+        fn roc__forHost_1_exposed_generic(_: *mut u8);
+        fn roc__forHost_1_exposed_size() -> usize;
+        fn roc__forHost_0_caller(flags: *const (), captures: *mut u8, model: *mut u8);
+        fn roc__forHost_0_size() -> usize;
+    }
+
+    unsafe {
+        let captures_size = roc__forHost_1_exposed_size();
+        let mut captures = RocList::from_raw_parts(
+            roc_alloc(captures_size, 0) as *mut u8,
+            captures_size,
+            captures_size,
+        );
+
+        let model_size = roc__forHost_0_size();
+        let mut model =
+            RocList::from_raw_parts(roc_alloc(model_size, 0) as *mut u8, model_size, model_size);
+
+        roc__forHost_1_exposed_generic(captures.as_mut_ptr());
+        roc__forHost_0_caller(
+            // This flags pointer will never get dereferenced
+            std::mem::MaybeUninit::uninit().as_ptr(),
+            captures.as_mut_ptr(),
+            model.as_mut_ptr(),
+        );
+
+        model.set_readonly();
+        captures.set_readonly();
+
+        RocServer {
+            model: model.into(),
+            captures: captures.into(),
```

Comment:
You can't save captures and keep reusing it like this. It isn't valid. This only works for `model` cause `model` is a `Box something`.

You need to do the same thing that we do in wasm4: https://github.com/lukewilliamboswell/roc-wasm4/blob/a216eb96e5d05dc2258cbf3da74facf787e93bcc/platform/host.zig#L133-L137